example/cajita_tutorial/08_array/array_example.cpp

```diff
@@ -0,0 +1,232 @@
+        std::cout << array_node_shared_owned_space.min( d ) << " ";
+    std::cout << "\nMax: ";
+    for ( int d = 0; d < dofs_per_node; ++d )
+        std::cout << array_node_shared_owned_space.max( d ) << " ";
+    std::cout << "\n" << std::endl;
+
+    /*
+      Next, we create an array from the array layout. This directly stores the
+      Kokkos View with field data that can be extracted for parallel
+      computation.
+    */
+    std::string label( "example_array" );
+    auto array = Cajita::createArray<double, device_type>( label, node_layout );
+
+    auto view = array->view();
+    std::cout << "Array total size: " << view.size() << std::endl;
+    std::cout << "Array extents: ";
+    for ( int i = 0; i < 4; ++i )
+        std::cout << view.extent( i ) << " ";
+    std::cout << "\n" << std::endl;
+
+    /*
+      Many array-based operations are available for convenience:
+       - assign a value to every field on the mesh
+       - scale the entire array
+       - update
+       - dot product between two arrays
+       - two-norm
+       - one-norm
+       - infinity-norm
+
+      For each operation we print the first value in the array - each value is
+      updated.
+    */
+    Cajita::ArrayOp::assign( *array, 2.0, Cajita::Ghost() );
+
+    // Scale the entire array with a single value.
+    Cajita::ArrayOp::scale( *array, 0.5, Cajita::Ghost() );
+
+    // Compute the dot product of the two arrays.
+    std::vector<double> dots( dofs_per_node );
+    auto array_2 =
+        Cajita::createArray<double, device_type>( label, node_layout );
+    Cajita::ArrayOp::assign( *array_2, 0.5, Cajita::Ghost() );
+    Cajita::ArrayOp::update( *array, 3.0, *array_2, 2.0, Cajita::Ghost() );
+    Cajita::ArrayOp::dot( *array, *array_2, dots );
+    std::cout << "Array dot product: ";
+    std::cout << dots[0] << std::endl;
+
+    // Compute the two-norm of the array components
+    std::cout << "Array two-norm: ";
+    std::vector<double> norm_2( dofs_per_node );
+    Cajita::ArrayOp::norm2( *array, norm_2 );
+    std::cout << norm_2[0] << std::endl;
+
+    // Compute the one-norm of the array components
+    std::cout << "Array one-norm: ";
+    std::vector<double> norm_1( dofs_per_node );
+    Cajita::ArrayOp::norm1( *array, norm_1 );
+    std::cout << norm_1[0] << std::endl;
+
+    // Compute the infinity-norm of the array components
+    std::cout << "Array infinity-norm: ";
+    view = array->view();
+    std::vector<double> large_vals = { -1939304932.2, 20399994.532,
+                                       9098201010.114, -89877402343.99 };
+    for ( int n = 0; n < dofs_per_node; ++n )
+        view( 4, 4, 4, n ) = large_vals[n];
+    std::vector<double> norm_inf( dofs_per_node );
+    Cajita::ArrayOp::normInf( *array, norm_inf );
+    std::cout << norm_inf[0] << std::endl;
+
+    // Check the copy.
+    Cajita::ArrayOp::copy( *array, *array_2, Cajita::Own() );
+
+    // Now make a clone and copy.
```

Comment:
Cloning an array makes a new, identical array to the given array with a new memory allocation.